ap/selection/test.py

```diff
@@ -1,306 +1,467 @@
+
+        # construct metric table for these objects. The metric table contains
+        # the minimal value of the metric *metric* for each object in field
+        # *to_clean* w.r.t. all objects in *summed_clean_against*. Thus,
+        # it has the dimensions nevents x nto_clean, where *nevents* is
+        # the number of events in the current chunk of data and *nto_clean*
+        # is the length of the field *to_clean*. Note that the argument
+        # *threshold* in the *nearest* function must be set to None since
+        # the function will perform a selection itself to extract the nearest
+        # objects (i.e. applies the selection we want here in reverse)
+        _, metric = to_clean_field.nearest(
+            summed_clean_against,
+            metric=metric,
+            return_metric=True,
+            threshold=None,
+        )
+        # build a binary mask based on the selection threshold provided by the
+        # user
+        mask = metric > threshold
+
+        # construct final result. Currently, this is the list of indices for
+        # clean jets, sorted for pt
+        # WARNING: this still contains the bug with the application of the mask
+        #           which will be adressed in a PR in the very near future
+        # TODO: return the mask itself instead of the list of indices
+        sorted_list = ak.argsort(to_clean_field.pt, axis=-1, ascending=False)[mask]
+        return sorted_list
+
+    return func
+
+
+delta_r_jet_lepton = cleaning_factory(
+    selector_name="delta_r_jet_lepton",
+    to_clean="Jet",
+    clean_against=["Muon", "Electron"],
+    metric=lambda a, b: a.delta_r(b),
+)
+
+
+def req_clean_delta_r_match_jet_lepton(
+    events: ak.Array,
+    to_clean: str,
+    clean_against: List[str],
+    threshold: float = 0.4,
+) -> List[int]:
+    """
+    do the cleaning of jets with respect to leptons.
+    returns indices of good (clean) jets, sorted by pt
+    *TODO*: this should probably changed to return the boolean masks
+    """
+    return delta_r_jet_lepton(events, to_clean, clean_against, threshold=threshold)
+
+
+@selector(uses={delta_r_jet_lepton})
+def jet_lepton_delta_r_cleaning_selection(
+    events: ak.Array,
+    stats: Dict,
+    threshold: float = 0.4,
+) -> SelectionResult:
+    """
+    function to apply the selection requirements necessary for a
+    cleaning of jets against leptons.
+    The function calls the requirements to clean the field *Jet* against
+    the concatination of the fields *[Muon, Electron]*, i.e. all leptons
+    and passes the desired threshold for the selection
+    """
+    clean_jet_indices = req_clean_delta_r_match_jet_lepton(events, "Jet", ["Muon", "Electron"], threshold=threshold)
+
+    return SelectionResult(
+        objects={"Jet": clean_jet_indices},
+    )
+
+
+@selector(uses={jet_lepton_delta_r_cleaning_selection})
+def jet_lepton_delta_r_cleaning(
+    events: ak.Array,
+    stats: Dict,
+    threshold: float = 0.4,
+    **kwargs: Dict,

```

Comment:
```suggestion
    stats: dict,
    threshold: float = 0.4,
    **kwargs,
```

I think `kwargs` don't need hints.

Reply:
```suggestion
    stats: Dict[str, Union[int, float]],
    threshold: float = 0.4,
    **kwargs,
```